code/modules/atmospherics/machinery/components/fusion/hfr_main_processes.dm

```diff
@@ -182,83 +42,65 @@
-		return
-
-	if(!selected_fuel)
-		return
-
-	if(!check_fuel())
-		return
-
-	if(!check_power_use())
+	if (check_power_use())
+		if (start_cooling)
+			inject_from_side_components(delta_time)
+			process_internal_cooling(delta_time)
+	else
+		// No power forces bad settings
 		magnetic_constrictor = 100
 		heating_conductor = 500
 		current_damper = 0
 		fuel_injection_rate = 20
 		moderator_injection_rate = 50
 		waste_remove = FALSE
-		iron_content += 0.1 * delta_time
+		iron_content += 0.02 * power_level * delta_time
+
+	update_temperature_status()
 
 	//Store the temperature of the gases after one cicle of the fusion reaction
 	var/archived_heat = internal_fusion.temperature
 	//Store the volume of the fusion reaction multiplied by the force of the magnets that controls how big it will be
 	var/volume = internal_fusion.volume * (magnetic_constrictor * 0.01)
 
-	//Assert the gases that will be used/created during the process
-	for(var/gas_id in selected_fuel.requirements) //These are the fuels
-		internal_fusion.assert_gas(gas_id)
+	//Store the fuel gases and the product gas moles
 
-	for(var/gas_id in selected_fuel.primary_products) //These are the gases that can be produced inside the internal_fusion mix
-		internal_fusion.assert_gas(gas_id)
+	var/energy_concentration_multiplier = 1
+	var/positive_temperature_multiplier = 1
+	var/negative_temperature_multiplier = 1
+	var/list/fuel_list = list()
 
-	internal_fusion.assert_gas(/datum/gas/antinoblium)
+	//We scale it down by volume/2 because for fusion conditions, moles roughly = 2*volume, but we want it to be based off something constant between reactions.
+	var/scale_factor = volume * 0.5
 
-	for(var/gas_id in GLOB.meta_gas_info)
-		moderator_internal.assert_gas(gas_id)
+	//Scaled down moles of gases, no less than 0
+	var/list/scaled_fuel_list = list()
 
-	//Store the fuel gases and the product gas moles
+	if (selected_fuel)
+		energy_concentration_multiplier = selected_fuel.energy_concentration_multiplier
+		positive_temperature_multiplier = selected_fuel.positive_temperature_multiplier
+		negative_temperature_multiplier = selected_fuel.negative_temperature_multiplier
 
-	var/list/fuel_list = list()
-	for(var/gas_id in selected_fuel.requirements)
-		fuel_list |= gas_id
-		fuel_list[gas_id] = internal_fusion.gases[gas_id][MOLES]
+		for(var/gas_id in selected_fuel.requirements)
+			fuel_list[gas_id] = internal_fusion.gases[gas_id][MOLES]
 
-	for(var/gas_id in selected_fuel.primary_products)
-		fuel_list |= gas_id
-		fuel_list[gas_id] = internal_fusion.gases[gas_id][MOLES]
+		for(var/gas_id in selected_fuel.primary_products)
+			fuel_list[gas_id] = internal_fusion.gases[gas_id][MOLES]
+
+		for(var/gas_id in selected_fuel.requirements)
+			scaled_fuel_list[gas_id] = max((fuel_list[gas_id] - FUSION_MOLE_THRESHOLD) / scale_factor, 0)
+
+		for(var/gas_id in selected_fuel.primary_products)
+			scaled_fuel_list[gas_id] = max((fuel_list[gas_id] - FUSION_MOLE_THRESHOLD) / scale_factor, 0)
```

Comment:
Tho I wonder, you could just throw this all in one for loop couldn't you?

Reply:
What did you have in mind? Is it efficient to iterating over a concatenated xvalue?

Reply:
Efficiency isn't hyper high priority here. Trust me 1 object iterating over gas types isn't the source of all our woes.

If I were going to do this I'd do something like
`for(var/gas_id in (selected_fuel.requirements | selected_fuel.primary_products)` 
then do the fuel_list and scaled_fuel_list stuff in the same loop. Might make things more unclear? IDK might help a bit

Reply:
I'll give it a shot and see how it looks

Reply:
```dm
for(var/gas_id in selected_fuel.requirements | selected_fuel.primary_products)
	var/amount = internal_fusion.gases[gas_id][MOLES]
	fuel_list[gas_id] = amount
	scaled_fuel_list[gas_id] = max((amount - FUSION_MOLE_THRESHOLD) / scale_factor, 0)
```

Seems good